### DISCUSSION TITLE
Self hosting with Docker

### DISCUSSION THREAD

**Discussion by RickCogley** [outline/outline#1796] [Self Hosting] [answered]
Hello - I'm considering self hosting, even though I think the cloud version pricing is more than fair, because I want to use my own S3 bucket to allow me to use a lambda function to optimize or resize any images uploaded to it. 

I am a newbie at Docker though, despite knowing system administration. Per [this issue](https://github.com/outline/outline/issues/1491) I can see I can deploy with Docker, but, what I'm not so clear on is: 

* is there a big performance hit using Docker? From reading various articles it appears "not very much". 
* if I were to use a droplet from the Digital Ocean market place that has Docker and Docker compose, does the `make up` command actually install redis and postgres, or, do I need to provide those in the DO server alongside docker, say with `apt get`? 
* along the same lines, do I need to install my own nginx to reverse proxy, as well, or does that automatically get installed?

Appreciate any advice, thanks. :-)

**Comment by tommoor** [comment, marked as answer]
Hi Rick,

1. There won't be any noticeable performance impact from running the application itself inside of Docker and it's definitely the recommended and simplest approach (it makes upgrading very simple).
2. `make up` and `docker-compose` really exists for development – in production you may setup postgres and redis however you see fit – you can certainly use their official Docker images, install them on the box directly, or use a hosted service.
3. Yes, if you want to expose the service to the wider internet with SSL you'd need to install a reverse proxy and configure it. We don't have documentation on this as there are just too many possible ways to set this up in the wild – I believe you could also use a hosted service such as Cloudflare for this too.

**Reply by RickCogley** [reply]
Very clear, thank you very much!

**Reply by nooobcoder** [reply]
@tommoor one thing i need to ask, is the pricing plan also applicable for the self hosted version?
I mean if i self host outline do I need to pay any subscription fees after 30 days of trial?
If so please confirm, I would disregard using outline instantly, because I wont risk my notes, data in it if it charges me after 30 days of usage in self hosted version. Thank You!

**Reply by tommoor** [reply]
No charges or subscription for self hosting
